app/javascript/orangelight/availability_show.js

```diff
@@ -0,0 +1,305 @@
+        }
+      })
+      .catch((error) => {
+        console.error(
+          `Failed to retrieve availability data for the bib. record ${this.id}: ${error.message}`
+        );
+      });
+  }
+
+  /* example with three host ids: https://bibdata.princeton.edu/bibliographic/availability.json?deep=true&bib_ids=9923427953506421,99125038613506421,99125026373506421,99124945733506421 */
+  // the record id is 9923427953506421
+  availability_url_show() {
+    let url = `${this.bibdata_base_url}/bibliographic/availability.json?deep=true&bib_ids=${this.id}`;
+    if (this.host_id.length > 0) {
+      const hostIds = Array.isArray(this.host_id)
+        ? this.host_id.join(',')
+        : this.host_id;
+      url += `,${hostIds}`;
+    }
+    return url;
+  }
+
+  // process_single() is used in the Show page and typically `holding_records` only has the
+  // information for a single bib since we are on the Show page. But occasionally the record
+  // that we are showing is bound with another (host) record and in those instances
+  // `holding_records` has data for two or more bibs: `this.id`, `this.host_id`.
+  // show page
+  process_single(holding_records) {
+    this.update_single(holding_records, this.id);
+    // Availability response in bibdata should be refactored not to include the host holdings under the mms_id of the record page.
+    // problematic availability response behavior for constituent record page with host records.
+    // It treats host records as holdings of the constituent record. see: https://github.com/pulibrary/bibdata/issues/1739
+    if (this.host_id.length > 0) {
+      const hostIds = Array.isArray(this.host_id)
+        ? this.host_id
+        : [this.host_id];
+      hostIds.forEach((mms_id) => {
+        this.update_single(holding_records, mms_id);
+      });
+    }
+  }
+
+  // show page
+  update_single(holding_records, id) {
+    return (() => {
+      const result = [];
+      for (const holding_id in holding_records[id]) {
+        const availability_info = holding_records[id][holding_id];
+        const { label } = holding_records[id][holding_id];
+        // case :constituent with host ids.
+        // data-record-id has a different this.id when there are host ids.
+        let availability_element;
+
+        // If we are not getting holding info select the availability element by record id only.
+        if (holding_id == 'RES_SHARE$IN_RS_REQ') {
+          availability_element = document.querySelector(
+            `*[data-availability-record='true'][data-record-id='${id}'][data-temp-location-code='RES_SHARE$IN_RS_REQ'] .availability-icon`
+          );
+        } else {
+          availability_element = document.querySelector(
+            `*[data-availability-record='true'][data-record-id='${id}'][data-holding-id='${holding_id}'] .availability-icon`
+          );
+          console.log(availability_element);
+        }
+        if (label) {
+          const holding_location = document.querySelector(
```

Comment:
Currently this is not working in production . [orangelight/issues/5151](https://github.com/pulibrary/orangelight/issues/5151)

Reply:
Ok to do this as separate PR since it involves the vue component.